src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParserStateObject.netcore.cs

```diff
@@ -1793,6 +1826,16 @@ private void SetBufferSecureStrings()
                             str = Marshal.SecureStringToBSTR(_securePasswords[i]);
                             byte[] data = new byte[_securePasswords[i].Length * 2];
                             Marshal.Copy(str, data, 0, _securePasswords[i].Length * 2);
+                            if (!BitConverter.IsLittleEndian)
+                            {
+                                byte temp;
+                                for (int ii = 0; ii < _securePasswords[i].Length * 2; ii += 2)
+                                {
+                                    temp = (byte)data[ii];
+                                    data[ii] = (byte)data[ii + 1];
+                                    data[ii + 1] = (byte)temp;
+                                }
+                            }
```

Comment:
Have you tested this? I wouldn't expect to need to byteswap protocol fields, the spec usually defines a specific endianness for them.

Reply:
The SqlCredentialTest's from the manual test suite were failing. I assume the bytes of the password data were not rearranged before calling ObuscatePassword (I guress that's were all protocol related stuff is done and then written back into the packet), please correct me I'am wrong.